src/main/java/com/hcl/appscan/jenkins/plugin/builders/AppScanBuildStep.java

```diff
@@ -329,6 +331,10 @@ private void perform(Run<?,?> build, Launcher launcher, TaskListener listener) t
             progress.setStatus(new Message(Message.WARNING, Messages.warning_sca()));
         }
 
+        if(m_type.equals("Dynamic Analyzer") && !properties.containsKey(Scanner.PRESENCE_ID) && !ServiceUtil.isValidUrl(target, m_authProvider, m_authProvider.getProxy())) {
```

Comment:
The number of if/else conditions in the perform() method is getting quite large. It would be good to move that logic to a private validateScan() method for clarity.

Reply:
This will be addressed in an upcoming PR.